Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/AccountController.cs

```diff
@@ -265,29 +267,52 @@ public IActionResult ForgotPassword()
-                }
+                    var user = await _userManager.FindByNameAsync(model.Email);
+                    if (user == null)
+                    {
+                        @ViewData["EmailNotExist"] = StringConstant.EmailNotExists;
+                        return View();
+                    }
 
-                // Send an email with this link
-                var code = await _userManager.GeneratePasswordResetTokenAsync(user);
-                var resetPasswordLink = Url.Action("ResetPassword", "Account", new { userId = user.Id, code = code }, protocol: HttpContext.Request.Scheme);
-                string path = _hostingEnvironment.ContentRootPath + StringConstant.ForgotPasswordTemplateFolderPath;
-                if (System.IO.File.Exists(path))
+                    // Send an email with this link
+                    var code = await _userManager.GeneratePasswordResetTokenAsync(user);
+                    var resetPasswordLink = Url.Action("ResetPassword", "Account", new { userId = user.Id, code = code }, protocol: HttpContext.Request.Scheme);
+                    string path = _hostingEnvironment.ContentRootPath + StringConstant.ForgotPasswordTemplateFolderPath;
+                    if (System.IO.File.Exists(path))
+                    {
+                        string finaleTemplate = System.IO.File.ReadAllText(path);
+                        finaleTemplate = finaleTemplate.Replace(StringConstant.ResetPasswordLink, resetPasswordLink).Replace(StringConstant.ResertPasswordUserName, user.FirstName);
+                        _emailSender.SendEmail(model.Email, StringConstant.ForgotPassword, finaleTemplate);
+                        @ViewData["MailSentSuccessfully"] = StringConstant.SuccessfullySendMail.Replace("{{emailaddress}}", "'" + model.Email + "'");
+                    }
+                }
+                // If we got this far, something failed, redisplay form
+                return View(model);
+            }
+            catch (InvalidApiRequestException apiEx)
+            {
+                _logger.LogInformation("Forgot Password mail not send " + apiEx.Message + apiEx.ToString());
+                if (apiEx.Errors.Count() > 0)
                 {
-                    string finaleTemplate = System.IO.File.ReadAllText(path);
-                    finaleTemplate = finaleTemplate.Replace(StringConstant.ResetPasswordLink, resetPasswordLink).Replace(StringConstant.ResertPasswordUserName, user.FirstName);
-                    _emailSender.SendEmail(model.Email, StringConstant.ForgotPassword, finaleTemplate);
-                    @ViewData["MailSentSuccessfully"] = StringConstant.SuccessfullySendMail.Replace("{{emailaddress}}", "'" + model.Email + "'");
+                    foreach (var error in apiEx.Errors)
+                        _logger.LogInformation("Forgot Password mail not send " + error);
                 }
+                throw apiEx;
+            }
+            catch (ArgumentNullException argEx)
```

Comment:
Because in Repository we have checked whether the SendGridApi and the From address is null or empty. If they are it means that arguments were null and ArgumentNullException have been thrown. Thus caught ArgumentNullException.